### PR TITLE
Nb/feat/drift scores

### PR DESCRIPTION
**Pull request by Nadav-Barak**
Resolves https://github.com/deepchecks/private-issues/issues/35

Implemented reduce_output() in drift checks.

### CODE REVIEW

File: deepchecks/vision/checks/train_test_validation/train_test_label_drift.py [hunk 1]
```diff
@@ -228,6 +229,11 @@ def compute(self, context: Context) -> CheckResult:
 
         return CheckResult(value=values_dict, display=displays, header='Train Test Label Drift')
 
+    def reduce_output(self, check_result: CheckResult) -> Dict[str, float]:
+        """Return label drift score per label property."""
+        drift_values = [label_property['Drift score'] for label_property in check_result.value.values()]
```
Comment (JKL98ISR) [inline deepchecks/vision/checks/train_test_validation/train_test_label_drift.py:234]: why not
{name: label_property['Drift score'] for name, label_property in check_result.value.items()}

**Review by JKL98ISR — APPROVED** [review]

**Review by ItayGabbay — APPROVED** [review]

File: deepchecks/tabular/checks/train_test_validation/train_test_feature_drift.py [hunk 2]
```diff
@@ -70,6 +74,13 @@ class TrainTestFeatureDrift(TrainTestCheck):
     categorical_drift_method: str, default: "cramer_v"
         decides which method to use on categorical variables. Possible values are:
         "cramers_v" for Cramer's V, "PSI" for Population Stability Index (PSI).
+    aggregation_method: str, default: "weighted"
+        decides how to aggregate the drift scores for a collective score. Possible values are:
```
Comment (noamzbr) [inline deepchecks/tabular/checks/train_test_validation/train_test_feature_drift.py:78]: Rephrase a bit to make it clear it's relevant to the reduce method

File: deepchecks/tabular/checks/train_test_validation/train_test_feature_drift.py [hunk 3]
```diff
@@ -70,6 +74,13 @@ class TrainTestFeatureDrift(TrainTestCheck):
...
+        'weighted': Weighted mean based on feature importance, provides a robust estimation on how
+        much the drift will affect the model's performance.
+        'mean': Simple mean of all drift scores.
```
Comment (noamzbr) [inline deepchecks/tabular/checks/train_test_validation/train_test_feature_drift.py:81]:
```suggestion
        'mean': Mean of all drift scores.
```

File: deepchecks/tabular/checks/train_test_validation/train_test_feature_drift.py [hunk 4]
```diff
@@ -220,6 +233,27 @@ def run_logic(self, context: Context) -> CheckResult:
...
+        drift_values = [col['Drift score'] for col in check_result.value.values()]
+        feature_names = list(check_result.value.keys())
+        if self.aggregation_method == 'none':
+            return dict(zip(feature_names, drift_values))
+        elif self.aggregation_method == 'mean':
+            return {'Mean Drift Score': np.mean(drift_values)}
+        elif self.aggregation_method == 'max':
+            return {'Max Drift Score': np.max(drift_values)}
+        elif self.aggregation_method == 'weighted':
+            feature_importance = [col['Importance'] for col in check_result.value.values()]
```
Comment (noamzbr) [inline deepchecks/tabular/checks/train_test_validation/train_test_feature_drift.py:247]:
```suggestion
            feature_importance = [col['Importance'] for col in feature_names]
```

File: docs/source/checks/tabular/train_test_validation/plot_train_test_feature_drift.py [hunk 5]
```diff
@@ -156,3 +157,27 @@
 #%%
 # As we see, our condition successfully detects and filters the problematic
 # features that contains a drift!
+#
+# Get an aggregated value
+# =======================
+# Using the reduce_output function we can aggregate the drift values per feature and get a collective score
+# that reflects the affect of the drift across all features on the model. In an unlabeled environment this value
```
Comment (noamzbr) [inline docs/source/checks/tabular/train_test_validation/plot_train_test_feature_drift.py:164]:
```suggestion
# that reflects the effect of the drift across all features on the model. In an unlabeled environment this value
```

File: docs/source/checks/tabular/train_test_validation/plot_train_test_feature_drift.py [hunk 6]
```diff
@@ -156,3 +157,27 @@
...
+# can be a good indicator of possible deterioration in the model performance.
+#
+# We can define the type of aggregation we want to use via the `aggregation_method` parameter. Its possible values are:
```
Comment (noamzbr) [inline docs/source/checks/tabular/train_test_validation/plot_train_test_feature_drift.py:167]:
```suggestion
# We can define the type of aggregation we want to use via the `aggregation_method` parameter. The possible values are:
```

**Review by noamzbr — COMMENTED** [review]
Shouldn't we also have a reduce in the property drift check?

File: deepchecks/tabular/checks/train_test_validation/train_test_feature_drift.py (hunk 4 above)
Comment (Nadav-Barak) [reply deepchecks/tabular/checks/train_test_validation/train_test_feature_drift.py:247]: Not the same

**Review by noamzbr — APPROVED** [review]

File: docs/source/checks/tabular/train_test_validation/plot_train_test_feature_drift.py [hunk 7]
```diff
@@ -156,3 +157,27 @@
 #%%
 # As we see, our condition successfully detects and filters the problematic
 # features that contains a drift!
+#
+# Get an aggregated value
+# =======================
+# Using the reduce_output function we can aggregate the drift values per feature and get a collective score
```
Comment (shir22) [inline docs/source/checks/tabular/train_test_validation/plot_train_test_feature_drift.py:163]: make the `reduce_output` a link to the api reference

Comment (shir22) [reply docs/source/checks/tabular/train_test_validation/plot_train_test_feature_drift.py:163]: Wouldn't repeat the "aggregated" in title and first sentence
(maybe change one of the to "single" or an alternative)

File: docs/source/checks/tabular/train_test_validation/plot_train_test_feature_drift.py [hunk 8]
```diff
@@ -156,3 +157,27 @@
...
+# that reflects the effect of the drift across all features on the model. In an unlabeled environment this value
```
Comment (shir22) [inline docs/source/checks/tabular/train_test_validation/plot_train_test_feature_drift.py:164]:
```suggestion
# that reflects the effect of the drift on the model, taking into account all of the features. When we don't have labels, this value
```

Comment (shir22) [reply docs/source/checks/tabular/train_test_validation/plot_train_test_feature_drift.py:164]: unlabeled environment sounds as if "we never get labels"
I wanted the wording to be clear that it's an environment in which we "currently don't have labels" (whether they will or will not arrive). Feel free to change it to something that will emphasize that more...

File: docs/source/checks/tabular/train_test_validation/plot_train_test_feature_drift.py [hunk 9]
```diff
@@ -156,3 +157,27 @@
...
+# can be a good indicator of possible deterioration in the model performance.
```
Comment (shir22) [inline docs/source/checks/tabular/train_test_validation/plot_train_test_feature_drift.py:165]:
```suggestion
# can be a good indicator of possible deterioration in the model's performance.
```

File: docs/source/checks/tabular/train_test_validation/plot_train_test_feature_drift.py [hunk 10]
```diff
@@ -156,3 +157,27 @@
...
+#
+# ``weighted``: The default. Weighted mean of drift scores based on each feature's feature importance. This method
+# underlying logic is that drift in a feature with a higher feature importance will have a greater effect on the model's
+# performance.
+#
+# ``mean``: Simple mean of all the features drift scores.
+#
+# ``none``: No averaging. Return a dict with a drift score for each feature.
+#
+# ``max``: Maximum of all the features drift scores.
```
Comment (shir22) [inline docs/source/checks/tabular/train_test_validation/plot_train_test_feature_drift.py:177]:
```suggestion
# ``max``: Maximum value of all the individual feature's drift scores.
```

File: docs/source/checks/tabular/train_test_validation/plot_train_test_feature_drift.py [hunk 11]
```diff
@@ -156,3 +157,29 @@
 #%%
 # As we see, our condition successfully detects and filters the problematic
 # features that contains a drift!
+#
+# Get an aggregated value
+# =======================
+# Using the :func:`deepchecks.tabular.checks.train_test_validation.TrainTestFeatureDrift.reduce_output` function we
+# can combine the drift values per feature and get a collective score
+# that reflects the effect of the drift on the model, taking into account all the features.
+# In scenarios where labels are unavailable (either temporary of permanently)
```
Comment (shir22) [inline docs/source/checks/tabular/train_test_validation/plot_train_test_feature_drift.py:166]:
```suggestion
# In scenarios where labels are unavailable (either temporarily of permanently)
```

File: docs/source/checks/tabular/train_test_validation/plot_train_test_feature_drift.py [hunk 12]
```diff
@@ -160,9 +160,11 @@
 #
 # Get an aggregated value
 # =======================
-# Using the reduce_output function we can aggregate the drift values per feature and get a collective score
-# that reflects the effect of the drift across all features on the model. In an unlabeled environment this value
-# can be a good indicator of possible deterioration in the model performance.
+# Using the :func:`deepchecks.tabular.checks.train_test_validation.TrainTestFeatureDrift.reduce_output` function we
```
Comment (shir22) [inline docs/source/checks/tabular/train_test_validation/plot_train_test_feature_drift.py:163]: add new line between title and text (rst best practice)

**Review by shir22 — APPROVED** [review]

**Pull request merged**